### ISSUE TITLE
BrowserWindow.getNativeWindowHandle() always returns a null NSView* on macOS

### ISSUE THREAD

**Issue by nikolovp** [electron/electron#15489]
* Output of `node_modules/.bin/electron --version`:
v3.0.6
v4.0.0-beta.5

* Operating System (Platform and Version):
macOS High Siearra 10.13.6

* Output of `node_modules/.bin/electron --version` on last known working Electron version (if applicable):
v2.0.11

**Expected Behavior**
BrowserWindow.getNativeWindowHandle() must return non-nullable result.

**Actual behavior**
BrowserWindow.getNativeWindowHandle() returns null.

**To Reproduce**

***Sample code***
```
const { app, BrowserWindow } = require('electron');

let win

app
	.on('ready', () => {
		win = new BrowserWindow({width: 800, height: 600})
		win.loadURL('https://www.google.com')
		
		win.on('closed', () => win = null);

		const handle = win.getNativeWindowHandle();
		console.log('Current window: ' + handle.toString('hex'));
	})
	.on('window-all-closed', () => app.quit());
```

***Expected output***
Something different than `Current window: 0000000000000000`

***Actual output***
`Current window: 0000000000000000`

**Comment by MarshallOfSound** [comment]
This appears to have been an accidental regression as a result of https://github.com/electron/electron/commit/4068a62fa629e6de3eea0ce64c425977ea320a28.

**Comment by Jun90925** [comment]
Same issue here on 3.0.8.
Which version will fix this issue？

**Comment by MarshallOfSound** [comment]
@Jun90925 There is a PR in flight for this issue #15521

**Comment by nikolovp** [comment]
Thanks for the fix! Can we expect it in one of the future v3.0.* patches?
